src/time/Sample.hpp

```diff
@@ -0,0 +1,14 @@
+
+#pragma once
+
+#include <Eigen/Core>
+
+namespace precice::time {
+
+/// @brief Sample containing data values
+struct Sample {
+  Eigen::VectorXd values;
+  Eigen::MatrixXd gradient;
```

Comment:
global data could then potentially also go here

Reply:
From the perspective of samples in waveforms: yes. But: The problem with global data that I see here is that global data does not have a mesh and gradient and value data have a mesh and currently samples are meant to live inside of data that is owned by a mesh.